tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -763,3 +803,68 @@ def test_positive_login_ad_user_basic_roles(
             ldapsession.usergroup.search('')
         ldapsession.architecture.create({'name': name})
         assert ldapsession.architecture.search(name)[0]['Name'] == name
+
+
+@upgrade
+@tier2
+def test_positive_login_user_password_otp(test_name, ipa_data, auth_source_ipa):
+    """Login with password with time based OTP
+
+    :id: be7eb5d6-3228-4660-aa64-c56f9f3ec5e0
+
+    :setup: Assure properly functioning IPA server for authentication
+
+    :steps: Login to server with an IPA user with time_based OTP.
+
+    :expectedresults: Log in to foreman UI successfully
+
+    """
+    try:
+        password_with_otp = "{0}{1}".format(
+            ipa_data['ldap_ipa_user_passwd'],
+            generate_otp(ipa_data['time_based_secret']))
+        with Session(
+                test_name,
+                ipa_data['ipa_otp_username'],
+                password_with_otp
+        ) as ldapsession:
+            with raises(NavigationTriesExceeded):
+                ldapsession.user.search('')
```

Comment:
I don't understand this. Why do we expect to see exception in positive path? Why do we search for empty string among users after logging in?

Reply:
There are multiple tests are using the same code. https://github.com/SatelliteQE/robottelo/blob/45fe1205f71a2956d6544cef8508360227f552b2/tests/foreman/ui/test_ldap_authentication.py#L777, https://github.com/SatelliteQE/robottelo/blob/45fe1205f71a2956d6544cef8508360227f552b2/tests/foreman/ui/test_ldap_authentication.py#L753 let me check what could be the exact reason.

Reply:
I guess the unless we fire any command e.g. ldapsession.user.search('') browser will not et open? that is only one reason I could guess

Reply:
let me check by removing that code as we have next step where we are reading the task details.

Reply:
Browser is opened on first request to entity, so in your case if you remove context manager and proceed directly to `ldapsession.task.read_all()`, it will still open browser.
Hacks like that would only be needed if you want to interact directly with `session.browser` before you interact with any entity (see "Browser needs to launch to grab datetime() for test case" on satqe-list posted around two weeks ago).

From what I see, in tests that you referenced this code tries to verify that logged in user doesn't have access to specific parts of Satellite. So that would be an answer to my question. Next thing is - do you need such assertion in your test? Seems you are more interested in login process itself than permissions after user logs in.

Reply:
my thought to verify that user got logged-in and has correct permissions.

Reply:
OK